pkg/services/system_intakes.go

```diff
@@ -141,27 +154,70 @@ func NewUpdateSystemIntake(
 					Model:     intake,
 					ModelID:   intake.ID.String(),
 					Operation: apperrors.Submit,
-					Source:    "CEDAR",
+					Source:    "CEDAR EASi",
 				}
 			}
 			intake.AlfabetID = null.StringFrom(alfabetID)
-		}
-		intake, err = update(intake)
-		if err != nil {
-			return &models.SystemIntake{}, &apperrors.QueryError{
-				Err:       err,
-				Model:     intake,
-				Operation: apperrors.QuerySave,
+			intake, err = update(intake)
+			if err != nil {
+				return &models.SystemIntake{}, &apperrors.QueryError{
+					Err:       err,
+					Model:     intake,
+					Operation: apperrors.QuerySave,
+				}
 			}
-		}
-		// only send an email when everything went ok
-		if intake.Status == models.SystemIntakeStatusSUBMITTED {
-			err = sendEmail(intake.Requester, intake.ID)
+			// only send an email when everything went ok
+			err = sendSubmitEmail(intake.Requester, intake.ID)
 			if err != nil {
 				return &models.SystemIntake{}, err
 			}
+
+			return intake, nil
+		} else if existingIntake.Status == models.SystemIntakeStatusSUBMITTED &&
```

Comment:
this is getting to be an unwieldy if case with the addition to this. Can we add abstract some of this into helper functions? Happy to talk about this more if you'd like.